### PR TITLE
Add informative error message if user forgets data arg in ggplot()

### PR DESCRIPTION
**Pull request by jrnold**
The error message `"gglot2 doesn't know how to deal with data of class uneval"` is one of the more common ones that I see, and one that perplexes new users of ggplot2. I added a note about the most common reason for and fix of this error message.

If a `data` argument was omitted from `ggplot()`, the old behavior was

```
> ggplot(aes(x = x))
Error: ggplot2 doesn't know how to deal with data of class uneval
```
The new behavior is
```
> ggplot(aes(x = x))
Error: ggplot2 doesn't know how to deal with data of class uneval. Did you forget the data argument  in ggplot()?
```

### CODE REVIEW

File: R/fortify.r [hunk 1]
```diff
@@ -19,9 +19,12 @@ fortify.NULL <- function(model, data, ...) waiver()
...
 #' @export
 fortify.default <- function(model, data, ...) {
-  stop(
+  msg <- paste0(
     "ggplot2 doesn't know how to deal with data of class ",
-    paste(class(model), collapse = "/"),
-    call. = FALSE
+    paste(class(model), collapse = "/"), "."
   )
+  if (class(model) == "uneval") {
```
Comment (hadley) [inline R/fortify.r]: This should use `inherits()`

File: R/fortify.r [hunk 2]
```diff
@@ -19,9 +19,12 @@ fortify.NULL <- function(model, data, ...) waiver()
...
+    msg <- paste0(msg, " Did you forget the data argument in ggplot()?")
```
Comment (hadley) [inline R/fortify.r]: Can you please surround `data` in backticks?

Is this function also called by geoms? If so, it might be confusing.

Comment (jrnold) [inline R/fortify.r]: I think that the error message will get called only if `ggplot()` has an `aes()` and no `data` argument:
```r
> ggplot() + geom_point(aes(x = x, y = y))
Error in FUN(X[[i]], ...) : object 'x' not found 
```
But 
```
> ggplot(aes(x = x, y = y)) + geom_point()
 Error: ggplot2 doesn't know how to deal with data of class uneval. Did you forget the `data` argument in ggplot()?
```

Comment (hadley) [inline R/fortify.r]: What about `geom_point(data = aes())` ?

Comment (jrnold) [inline R/fortify.r]: Good point. It does appear in that case. how about? 
```
Did you accidentally provide the results of `aes()` to the `data` argument?"
```

**Comment by hadley** [comment]
Looks like something has gone wrong with your merge/rebase

**Comment by jrnold** [comment]
I rewrote history so nothing bad ever happend

**Comment by hadley** [comment]
Thanks!

**Comment by jrnold** [comment]
👍

**Pull request merged**